src/ILCompiler.Compiler/src/Compiler/PreInitFieldInfo.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Debug = System.Diagnostics.Debug;
+
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler
+{
+    public class PreInitFieldInfo
+    {
+        public FieldDesc Field { private set; get; }
```

Comment:
Nit: you could remove the `private set` part to make the backing fields `readonly`.